### ISSUE TITLE
IOS fails to build because of unrelated OSX OpenGL components

### ISSUE THREAD

**Issue by nobody37** [liballeg/allegro5#680]
Using the example project for IOS included with Allegro it wouldn't build at first because I needed to include a preprocessor directive for ALLEGRO_IPHONE. Otherwise it would try to execute the code in #ifdef ALLEGRO_MACOSX.

Now it fails in gl_ext_api.h:
Unknown type name 'GLdouble' did you mean 'double'?
Redefinition of parameter 'GLdouble'

It has that error in multiple places.

HELP

**Comment by goobliata** [comment]
Looks to me like you've not got the right alplatf.h being included. Make sure you don't have the one from your Mac build lying around. It should be using the one in the directory beside the iOS Xcode project.

**Comment by nobody37** [comment]
Thanks goobliata

All references to alplatf.h are written as:
#include "allegro5/platform/alplatf.h"
They are in allegro.c, directx_shaders.cpp, inline.c, prim_directx.cpp, prim_opengl.c, primitives.c

I looked in my ~/Documents/Git/allegro/.../include/allegro5/platform and sure enough there is a #define ALLEGRO_MACOSX.

So does that mean that I need to modify my includes any time I need to switch between IOS and OSX builds?

When I try to run:
cmake -DCMAKE_TOOLCHAIN_FILE=../cmake/Toolchain-iphone.cmake -G Xcode \
>     -DIOS_PLATFORM="iphonesimulator" ..
At the command line it builds, but then when I run "xcodebuild" per the directions in README_iphone.txt I get:

In file included from /Users/john/Documents/Git/allegro/allegro/include/allegro5/allegro.h:73:
/Users/john/Documents/Git/allegro/allegro/include/allegro5/platform/alosx.h:37:12: fatal error: 'AppKit/AppKit.h' file not found
   #import <AppKit/AppKit.h>
           ^
It's still trying to do some OSX stuff in there. Any ideas? Sorry I'm new at this.

**Comment by goobliata** [comment]
Well where you put "..." in your path is important, it depends what that is. Your includes shouldn't interfere if you do it correctly. e.g. on my system I have allegro5/build where I have my macOS build which has include/allegro5/platform/alplatf.h. Then I use the iOS project to build for iOS. These don't interfere. The only way they would is if you copied the include files to some other directory or if you put extra include directories in the iOS Xcode project.

As for using CMake with iOS, that is sketchy. At the very least you'll need to open the Xcode project in Xcode and manually adjust a few things such making it build for iOS instead of Mac.

**Comment by nobody37** [comment]
This is my only copy of allegro on the system, pulled from Git:
The full file path is
/Users/john/Documents/Git/allegro/allegro/include/allegro5/platform/alplatf.h

Originally when building for OSX I did a make/make install after cmake. This copied the same header files into /usr/local/include/allegro5/platform/alplatf.h

Both files are the same contents. I got this version from Git (this is my first version of installing Allegro so no other previous installs exist).

What things do I need to adjust in Xcode?

Why is there a #define ALLEGRO_MACOSX in my alplatf.h file? Is that modified by the CMake process? (I'm new to CMake I'm used to doing configure/make/make install on builds of various sources). I'm going to assume that wasn't just hardcoded into the process by the developers, since I see this file here on Git:
https://github.com/liballeg/allegro5/blob/master/include/allegro5/platform/alplatf.h.cmake

So yes it's "sketchy" and frankly I don't like using CMake already, but it looks logically to me like it's necessary to get even a minimal IOS build. >_> 

What do I have to do to get this fixed, and why is CMake being a stubborn goat and trying to default to MACOSX (clearly since it's trying to find/use AppKit and not UIKit and burning in flames)?

**Comment by goobliata** [comment]
Well you shouldn't have alplatf.h in that directory in your Allegro git checkout, I don't know how it got there but it shouldn't be there. The one in /usr/local might or might not interfere, and yes as you said CMake generates alplatf.h so it was generated when you were building for Mac.

The iOS project included with Allegro will work better than CMake for iOS provided those alplatf.h files are not there. You still may need to edit the project to add paths to your dependencies but that's about it.

**Comment by nobody37** [comment]
Removing alplatf.h from the allegro directory (I didn't even have to modify /usr/local/include/allegro5/platform/altplatf.h) caused the CMAKE to build for IOS successfully!

So I think it was made the first time I ran CMAKE for OSX and the stale file was there, it needs to be deleted and regenerated to properly build IOS, I would probably make a note in the Readme docs of that. The IOS project wouldn't even build without it. 

The docs aren't very clear on how to get it actually on IOS, when I build it succeeds but since it's building a library it's not deploying automatically, it says something about copying libs and headers but it's not very clear and specific in how to do so. How do I get my code onto the simulator and onto my iPhone for running and testing?

**Comment by nobody37** [comment]
Closing this request now I've got it all working.

The steps to resolve were to build from cmake via 
cmake -DCMAKE_TOOLCHAIN_FILE=../cmake/Toolchain-iphone.cmake -G Xcode \
    -DIOS_PLATFORM="iphoneos" -DWANT_EXAMPLES=off -DWANT_DEMO=off -DWANT_TESTS=off ..
    xcodebuild
Then run ALLEGRO.xcodeproj in the build directory (not in misc/Allegro 5 iOS per the docs).
Then create a new IOS project, add all .a files from the initial build, and add the output paths to the Library and Header search paths.
I also deleted main.m and created a new main.c and al_init(); now runs.
